### PR TITLE
test: add migration edge case tests

### PR DESCRIPTION
**Pull request by mthalman**
Co-authored-by: Copilot <223556219+Copilot@users.noreply.github.com>

### CODE REVIEW

File: packages/core/src/test/migration.test.ts [hunk 1]
```diff
@@ -169,3 +169,254 @@ describe('Migration: existing WorkItems → accepted state', () => {
...
+      makeWorkItem({ id: 'a', providerId: 'gh', externalId: 'issue-1' }),
+      makeWorkItem({ id: 'b', providerId: 'gh', externalId: 'issue-2' }),
+      makeWorkItem({ id: 'c', providerId: 'gh', externalId: 'issue-3' }),
+    ];
+    const store = createMockStore(items);
+    const graph = new WorkGraph(store);
+    await graph.load();
+
+    // issue-2 already has state
+    stateStore.getState.mockImplementation((providerId: string, externalId: string) => {
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/core/src/test/migration.test.ts]: Fixed in 8aa76ea11a91b6978cc33b0cbd58a8dbc2473be9

File: packages/core/src/test/migration.test.ts [hunk 2]
```diff
@@ -169,3 +169,254 @@ describe('Migration: existing WorkItems → accepted state', () => {
...
+        });
+      }
+    }
+  }
+
+  if (itemsToMigrate.length > 0) {
+    try {
+      await stateStore.setStates(itemsToMigrate);
+    } catch {
+      // Matches extension.ts catch block — error is logged but not rethrown
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/core/src/test/migration.test.ts]: Fixed in 0ab6d82aebb4a66234721824a308e408ab38a207

File: packages/core/src/test/migration.test.ts [hunk 3]
```diff
@@ -169,3 +169,254 @@ describe('Migration: existing WorkItems → accepted state', () => {
...
+    loadAll: vi.fn(async () => []),
+    onDidChange: vi.fn(() => ({ dispose: vi.fn() })),
+    dispose: vi.fn(),
+  };
+}
+
+/**
+ * Replicates the actual extension.ts migration logic (lines 56-78) which
+ * collects items into an array and calls setStates once (batch), rather than
+ * calling setState individually per item.
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/core/src/test/migration.test.ts]: Fixed in 61a5cf38f692d91c710eb3f9c51c03da27ac025f

File: packages/core/src/test/migration.test.ts [hunk 4]
```diff
@@ -169,3 +169,254 @@ describe('Migration: existing WorkItems → accepted state', () => {
...
+      makeWorkItem({ id: 'b', providerId: 'gh', externalId: 'issue-2' }),
+      makeWorkItem({ id: 'c', providerId: 'gh', externalId: 'issue-3' }),
+    ];
+    const store = createMockStore(items);
+    const graph = new WorkGraph(store);
+    await graph.load();
+
+    // issue-2 already has state
+    stateStore.getState.mockImplementation((_providerId: string, externalId: string) => {
+      if (externalId === 'issue-2') {return 'accepted' as any;}
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/core/src/test/migration.test.ts]: Fixed in e53fe0df49bcb904aa62d738746a40c75aa12ec1
